### PR TITLE
ImageCarousel drag gesture and layout improvement 

### PR DESCRIPTION
**Pull request by adrianpdm**
### Task Description
Carousel can now bleed fully edge to-edge within a page.
Carousel is only draggable if total slide is greater than the configured slides per view.

### Changes
- Carousel is configurable to responsively bleed using `swiperProps` attribute.
- Automatically compute whether dragging is allowed by comparing expected slides per view `vs` total slides length
- Refactor navigation visibility handler

#### Before (Clipped Carousel)
Screenshot from 2021-09-10 10 36 13

#### After (Edge-to-edge Carousel)

https://user-images.githubusercontent.com/20709202/132795489-1f1c99d8-8919-479a-85ef-4898fb32cad5.mp4

### CODE REVIEW

**Review by Arif9878 — APPROVED** [review]

**Review by Ibwedagama — APPROVED** [review]

File: components/ImageCarouselV2/ImageCarousel.vue [hunk 1]
```diff
@@ -176,14 +189,52 @@ export default {
...
+          return window.innerWidth >= width
+        }
+        const [min, max] = [width, sorted[i - 1]]
+        return _inRange(window.innerWidth, min, max)
+      })
+    },
+    getConfigByBreakpoint (name) {
+      const bp = this.getMatchedBreakpoint()
+      let value = _get(this.swiperOptions, `breakpoints.${bp}.${name}`)
+      if (typeof value === 'undefined') {
```
Comment (bangunbagustapa) [inline components/ImageCarouselV2/ImageCarousel.vue:211]: Is this different from `if (!value)` ?

**Review by bangunbagustapa — APPROVED** [review]

**Review by maulanayuseph — APPROVED** [review]

**Review by adzharamrullah — APPROVED** [review]

File: components/ImageCarouselV2/ImageCarousel.vue [hunk 2]
```diff
@@ -176,14 +189,52 @@ export default {
     this.isServerSide = false
   },
   methods: {
-    onSwiperReady () {
+    getMatchedBreakpoint () {
+      // sorted and reversed due to
+      // min-width matching strategy
+      const sorted = [...this.swiperBreakpoints]
+        .sort((a, b) => a - b)
+        .reverse()
```
Comment (yoslie) [inline components/ImageCarouselV2/ImageCarousel.vue:197]: is reverse() to sort data in descending? why dont use .sort((a, b) => b - a) instead?

Comment (adrianpdm) [reply components/ImageCarouselV2/ImageCarousel.vue:197]: ... why didnt i think of that :open_mouth: 
thank you for your lit :fire: suggestion. on my way to fix it.

Comment (adrianpdm) [reply components/ImageCarouselV2/ImageCarousel.vue]: Hi @yoslie , i fixed it on this commit ([18b23ca](https://github.com/jabardigitalservice/pikobar-jabarprov-go-id/pull/361/commits/18b23cabe69da3108eec43a09f6cf94ddb878d68))

**Review by yoslie — APPROVED** [review]

File: components/ImageCarouselV2/ImageCarousel.vue (hunk 1 above)
Comment (adrianpdm) [reply components/ImageCarouselV2/ImageCarousel.vue:210]: Value can be falsy, e.g `0`, `false`, but never `undefined`.
When value is `undefined`, value will fallback from its breakpoint-specific config onto the global config.

For example:
```javascript
const config = {
  allowTouchMove: true,
  breakpoints: {
    480: {
      allowTouchMove: false
    }
  }
}
```

Looking into the case above, using negation would address the wrong value.

Comment (bangunbagustapa) [reply components/ImageCarouselV2/ImageCarousel.vue:210]: Ah i see, it's different case for `undefined`. Thanks for the explanation.

**Pull request merged**